### PR TITLE
ICMSLST-556 Add DLF Application Goods Certificates.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Features:
  - Add goods section to main form
  - List goods certificates
  - Add good certificate
  - Edit goods certificate
  - Delete goods certificate
  - View goods certificate

### CODE REVIEW

File: web/domains/case/_import/fa_dfl/models.py [hunk 1]
```diff
@@ -1,18 +1,63 @@
...
+class DFLGoodsCertificate(File):
+    """Deactivated Firearms Licence Goods certificate"""
+
+    goods_description = models.CharField(
+        max_length=4096, verbose_name="Goods Description", blank=False, null=False
+    )
+
+    deactivated_certificate_reference = models.CharField(
+        max_length=50, verbose_name="Deactivated Certificate Reference", blank=False, null=False
+    )
```
Comment (kevinetienne) [inline web/domains/case/_import/fa_dfl/models.py:18]: Are `blank=False` and `null=False` the default for django model fields? If yes maybe we might not need them.

File: web/domains/case/_import/fa_dfl/models.py [hunk 2]
```diff
@@ -1,18 +1,63 @@
...
+        dcf = self.deactivated_certificate_reference
+        return f"DFLGoodsCertificate(id={self.pk}, deactivated_certificate_reference={dcf!r})"
 
 
 class DFLApplication(ImportApplication):
     """Firearms & Ammunition Deactivated Firearms Licence application"""
 
+    class CommodityCodes(models.TextChoices):
+        EX_CHAPTER_93 = "ex Chapter 93", "ex Chapter 93"
+        EX_CHAPTER_97 = "ex Chapter 97", "ex Chapter 97"
```
Comment (kevinetienne) [inline web/domains/case/_import/fa_dfl/models.py:39]: Can you add the parenthesis for the tuples please?

File: web/domains/case/_import/models.py [hunk 3]
```diff
@@ -234,13 +238,15 @@ class ImportApplication(WorkbasketBase, Process):
...
         related_name="import_applications_from",
+        verbose_name="Country Of Origin",
     )
     consignment_country = models.ForeignKey(
         Country,
         on_delete=models.PROTECT,
         blank=True,
         null=True,
         related_name="import_applications_to",
+        verbose_name="Country Of Consignment",
```
Comment (kevinetienne) [inline web/domains/case/_import/models.py:249]: Does adding `verbose_name` require to update the migrations?

Comment (MattHolmes123) [reply web/domains/case/_import/models.py:249]: Ahh yes it does... are we ok with this?
I just noticed those labels were defined in 3 or so forms when in reality it should be the verbose_name

File: web/domains/case/_import/fa_dfl/forms.py [hunk 4]
```diff
@@ -79,5 +64,58 @@ def __init__(self, *args, **kwargs):
...
+            country_groups__name="Firearms and Ammunition (Deactivated) Issuing Countries"
+        ),
+    )
+
+
+class EditDLFGoodsCertificateForm(forms.ModelForm):
+    class Meta:
+        model = models.DFLGoodsCertificate
+        fields = ("goods_description", "deactivated_certificate_reference", "issuing_country")
+        help_texts = {
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_dfl/forms.py:107]: empty line above please

File: web/domains/case/_import/fa_dfl/models.py [hunk 5]
```diff
@@ -1,18 +1,63 @@
...
+from web.domains.constabulary.models import Constabulary
+from web.domains.country.models import Country
+from web.domains.file.models import File
+
+
+class DFLGoodsCertificate(File):
+    """Deactivated Firearms Licence Goods certificate"""
+
+    goods_description = models.CharField(
+        max_length=4096, verbose_name="Goods Description", blank=False, null=False
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_dfl/models.py:13]: how about it verbose name is exactly what django would create anyway we don't manually specify it

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**